Documentation/specs/corehost.md

```diff
@@ -0,0 +1,112 @@
+* `AssetName` -> The base name of the Asset. This is generally just the file name, without the extension (also removing "sub-extensions" like `ni` for native images of managed assemblies)
+* `AssetRelativePath` -> The relative path to the Asset file within the root of the Library.
+
+A few notes about the format:
+
+* The current code does not include the header, this is being changed.
+* All fields MUST be quoted (for simplicity), even if they don't contain a ','
+
+### Files in the application folder
+
+Any file with the suffix `.dll` in the same folder as the managed application being loaded (the "Application Base") will be considered a viable assembly during the resolution process. The host **assumes** that the assembly's short name is the same as the file name with the `.dll` suffix removed (yes, this is not technically required by the CLR, but we assume it for use with this host).
+
+### Files from package caches
+
+Only assemblies listed in the dependencies file can be resolved from a package cache. To resolve those assemblies, two environment variables are used:
+
+* `DOTNET_PACKAGES` - The primary package cache. If not set, defaults to `$HOME/.nuget/packages` on Unix or `%LOCALAPPDATA%\NuGet\Packages` (TBD) on Windows. **NOTE**: Currently the host uses different folders as we are still coordinating with NuGet to get the directories right (there are compat considerations). Currently we always use `$HOME/.dnx/packages`(Unix)/`%USERPROFILE%\.dnx\packages`(Win).
+* `DOTNET_PACKAGES_CACHE` - The secondary cache. This is used by shared hosters (such as Azure) to provide a cache of pre-downloaded common packages on a faster disk. If not set, it is not used.
+
+Given the Package ID, Package Version, Package Hash and Asset Relative Path provided in the dependencies file, resolution proceeds as follows (Unix-style paths will be used for convenience but these variables and paths all apply to Windows as well):
+
+1. If `DOTNET_PACKAGES_CACHE` is non-empty, read the file `$DOTNET_PACKAGES_CACHE/[Package ID]/[Package Version]/[Package Id].[Package Version].nupkg.sha512` if present. If the file is present and the content matches the `[Package Hash]` value from the dependencies file. Use that location as the Package Root and go to 3
+2. Using `DOTNET_PACKAGES`, or it's default value, use `$DOTNET_PACKAGES/[Package ID]/[Package Version]` as the Package Root
+3. Concatenate the Package Root and the Asset Relative Path. This is the path to the asset (managed assembly or native library).
+
+## Assembly Resolution
+
+During host start-up, the host identifies if a `.deps` file is present and loads it. It also scans the files located in the Application Base and determines the assembly name for each (using the file name). It builds a set of assembly names to be loaded by the union of the assembly names listed in `.deps` file and the assemblies located in the Application Base.
```

Comment:
It may be better to call out that in the absence of the deps file, servicing of local assembly is not possible without version and its original and relative servicing path.